File: CS/PropertyGridEditorAttribute/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.ObjectModel;

namespace PropertyGridEditor_Attribute {
    public class ViewModel : ViewModelBase {
        public Department Department { get; set; }
        public ViewModel() {
            Department = new Department {
                Name = "Marketing",
                Location = new Location { Country = "USA", City = "Seattle", Adress = "11th Ave. N.E.", ZIP = 4726 },
                Employees = new ObservableCollection<Employee>() {
                    new Employee { FirstName = "Jett", LastName = "Mitchell", BirthDate = new DateTime(1976,12,8), Married = true },
                    new Employee { FirstName = "Garrick", LastName = "Stiedemann", BirthDate = new DateTime(1985,1,9), Married = false },
                    new Employee { FirstName = "Hettie", LastName = "Runte", BirthDate = new DateTime(1981,7,2), Married = true }
                }
            };
        }
        public IDialogService EditEmployeeService { get { return GetService<IDialogService>(); } }

        [Command]
        public void EditEmployee(object item) {
            if (item != null && EditEmployeeService != null) {
                EditEmployeeService.ShowDialog(MessageButton.OK, "Edit Employee", item);
            }
        }

    }
}

File: VB/PropertyGridEditorAttribute/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Collections.ObjectModel

Namespace PropertyGridEditor_Attribute

    Public Class ViewModel
        Inherits ViewModelBase

        Public Property Department As Department

        Public Sub New()
            Department = New Department With {.Name = "Marketing", .Location = New Location With {.Country = "USA", .City = "Seattle", .Adress = "11th Ave. N.E.", .ZIP = 4726}, .Employees = New ObservableCollection(Of Employee)() From {New Employee With {.FirstName = "Jett", .LastName = "Mitchell", .BirthDate = New DateTime(1976, 12, 8), .Married = True}, New Employee With {.FirstName = "Garrick", .LastName = "Stiedemann", .BirthDate = New DateTime(1985, 1, 9), .Married = False}, New Employee With {.FirstName = "Hettie", .LastName = "Runte", .BirthDate = New DateTime(1981, 7, 2), .Married = True}}}
        End Sub

        Public ReadOnly Property EditEmployeeService As IDialogService
            Get
                Return GetService(Of IDialogService)()
            End Get
        End Property

        <Command>
        Public Sub EditEmployee(ByVal item As Object)
            If item IsNot Nothing AndAlso EditEmployeeService IsNot Nothing Then
                EditEmployeeService.ShowDialog(MessageButton.OK, "Edit Employee", item)
            End If
        End Sub
    End Class
End Namespace
